### PR TITLE
Add the ability to read COMSOL .mphtxt files.

### PR DESCRIPTION
**Pull request by bangerth**
$SUBJECT says it all. The format is not particularly difficult to read and parse, but it assigns indicators to interior faces and edges, and we don't gracefully handle this in `Triangulation::create_triangulation()` (for good reason) so one has to fix it up manually and that takes about as much code as just reading in the files.

/rebuild

### CODE REVIEW

File: source/grid/grid_in.cc [hunk 1]
```diff
@@ -1473,8 +1952,8 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
   else
     AssertThrow(false, ExcInvalidGMSHInput(line));
 
-  // if file format is 2.0 or greater then we also have to read the rest of the
-  // header
+  // if file format is 2.0 or greater then we also have to read the rest of
+  // the header
```
Comment (bangerth) [inline source/grid/grid_in.cc:1956]: I have to admit that I don't know what happened here, but the remainder of the changes in this file are just wrapped around comments.

File: tests/grid/grid_in_comsol_mphtxt_01.cc [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include "../tests.h"
+
+
+
+template <int dim>
+void
+gmsh_grid(const char *name)
```
Comment (Rombur) [inline tests/grid/grid_in_comsol_mphtxt_01.cc:32]: :)

**Review by Rombur — APPROVED** [review]
Looks good. There is just a typo in the test .

**Comment by bangerth** [comment]
Thanks, @Rombur ! I already have a couple of follow-up patches. Would you be willing to let me fix the two things you marked up in these follow-ups, to expedite the process?

**Comment by bangerth** [comment]
Thanks! Follow-up patch coming up shortly!

**Pull request merged**